packages/dht/src/dht/DhtNode.ts

```diff
@@ -112,24 +114,47 @@ const logger = new Logger(module)
 
 export type Events = TransportEvents & DhtNodeEvents
 
-export const createPeerDescriptor = (msg?: ConnectivityResponse, nodeId?: DhtAddress): PeerDescriptor => {
+const calculateNodeIdRaw = (ipAddress: number, privateKey: Uint8Array): DhtAddressRaw => {
+    // nodeId is calculated as 
+    // concatenate(get104leastSignificatBits(hash(ipAddress)), 
+    //  get56leastSignificatBits(sign(ipAddress)))
+
+    const ipAsBuffer = Buffer.alloc(4)
+    ipAsBuffer.writeUInt32BE(ipAddress)
+
+    const ipHash = hash(ipAsBuffer)
+    const signature = createSignature(ipAsBuffer, privateKey)
+
+    const nodeIdRaw = Buffer.concat([ipHash.slice(ipHash.length - 13, ipHash.length),
+        signature.slice(signature.length - 7, signature.length)])
+
+    return nodeIdRaw
+}
+
+export const createPeerDescriptor = (msg: ConnectivityResponse, nodeId?: DhtAddress): PeerDescriptor => {
+
+    const privateKey = crypto.randomBytes(32)
+    const publicKey = crypto.randomBytes(20)  // TODO calculate publicKey from privateKey
```

Comment:
Do we still need publicKey, see Petri's point. If we do, we need to see if we can recover it here without `ethers`.

Reply:
Move function

Reply:
Discussed. We do need the publicKey.